### PR TITLE
[4] Bunch of docblock updates

### PR DESCRIPTION
**Pull request by PhilETaylor**
### Summary of Changes

Just a spammy looking, sofa sitting, bunch of docblock updates based on what phpStorm was complaining about 

Mostly adding `|null` where a param has a default of null. 

### Testing Instructions

Review the docblocks ... that's all

**And for the record - please DONT add `hacktoberfext-accepted` to this PR - it was not intended as a spammy T-shirt earning PR**

### CODE REVIEW

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 0adce1b7522338180c91a6fd6561e9942fe60549<br /><br />I looked at the diff, found all the changed files and checked applying the patch made the changes indicated - all good.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30949">issues.joomla.org/tracker/joomla-cms/30949</a>.</sub>

File: administrator/components/com_categories/src/Controller/CategoryController.php [hunk 1]
```diff
@@ -39,15 +39,16 @@ class CategoryController extends FormController
...
+	 * @param   MVCFactoryInterface|null  $factory  The factory.
+	 * @param   CMSApplication|null       $app      The JApplication for the dispatcher
+	 * @param   Input|null                $input    Input
 	 *
 	 * @since  1.6
 	 * @see    \JControllerLegacy
+	 * @throws \Exception
 	 */
-	public function __construct($config = array(), MVCFactoryInterface $factory = null, $app = null, $input = null)
+	public function __construct($config = array(), MVCFactoryInterface $factory = null, CMSApplication $app = null, Input $input = null)
```
Comment (zero-24) [inline administrator/components/com_categories/src/Controller/CategoryController.php:51]: I'm not sure but could have that change here by adding type hints have any impact?

Comment (ceford) [reply administrator/components/com_categories/src/Controller/CategoryController.php:51]: With the patch applied, I created a new Category in contacts - worked. and edited a Category in content - also worked. Is that enough?

Comment (zero-24) [reply administrator/components/com_categories/src/Controller/CategoryController.php:51]: I mean more generally whether that could have an impact we might have to document.

Comment (laoneo) [reply administrator/components/com_categories/src/Controller/CategoryController.php:51]: I guess you mean in comparison to J3. Not really as we introduced the MVCFactoryInterface to the list of parameters anyway in J4. As long as it is optional we are save here.

Comment (zero-24) [reply administrator/components/com_categories/src/Controller/CategoryController.php:51]: Ok thanks :-)

**Comment by wilsonge** [comment]
Added the label just because you mentioned it ;) 

But thankyou!

**Comment by PhilETaylor** [comment]
> Added the label just because you mentioned it ;)

I have 56 PRs so far this month... but I still need to put clothes on my kids back... she is still sleeping in a 2018 hacktoberfest T-shirt!

**Pull request merged**